source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems,
+       Nix community has chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is Open Source. 
+          It's only possible to cross-compile between ``aarch64-darwin`` and ``x86_64-darwin``.
+
+
+What's a target platform?
+-------------------------
+
+There's actually a third platform named target.
+
+It matters in cases where you'd like to distribute a compiler binary, 
+as you'd then like to build a compiler on the build platform, compile code on the
+target plaform and run the final executable on the host platform.
+
+Since that's rarely needed, we'll treat target platform the same as the build.
+
+
+Determining the host platform
+-----------------------------
+
+The build platform is determined automatically by Nix
+as it can just guess it during the configure phase.
```

Comment:
You can also set it by passing `localSystem`.

Reply:
In what cases is that really needed? In most cases guessing should be sufficient?

Reply:
The two main use cases would be remote build (if you have a builder with a different platform than your local machine) and testing evaluation of cross-related things by simulating being on another platform.

Reply:
I'd like to mention that then in #157 tutorial (and consider developing nixpkgs out of scope here).